src/transformers/processing_utils.py

```diff
@@ -165,6 +246,154 @@ def save_pretrained(self, save_directory, push_to_hub: bool = False, **kwargs):
+                    " it from 'https://huggingface.co/models', make sure you don't have a local directory with the"
+                    f" same name. Otherwise, make sure '{pretrained_model_name_or_path}' is the correct path to a"
+                    f" directory containing a {PROCESSOR_NAME} file"
+                )
+
+        try:
+            # Load processor dict
+            with open(resolved_processor_file, "r", encoding="utf-8") as reader:
+                text = reader.read()
+            processor_dict = json.loads(text)
+
+        except json.JSONDecodeError:
+            raise EnvironmentError(
+                f"It looks like the config file at '{resolved_processor_file}' is not a valid JSON file."
+            )
+
+        if is_local:
+            logger.info(f"loading configuration file {resolved_processor_file}")
+        else:
+            logger.info(f"loading configuration file {processor_file} from cache at {resolved_processor_file}")
+
+        if "auto_map" in processor_dict and not is_local:
+            processor_dict["auto_map"] = add_model_info_to_auto_map(
+                processor_dict["auto_map"], pretrained_model_name_or_path
+            )
+
+        return processor_dict, kwargs
+
+    @classmethod
+    def from_args_and_dict(cls, args, processor_dict: Dict[str, Any], **kwargs):
+        """
+        Instantiates a type of [`~processing_utils.ProcessingMixin`] from a Python dictionary of parameters.
+
+        Args:
+            processor_dict (`Dict[str, Any]`):
+                Dictionary that will be used to instantiate the processor object. Such a dictionary can be
+                retrieved from a pretrained checkpoint by leveraging the
+                [`~processing_utils.ProcessingMixin.to_dict`] method.
+            kwargs (`Dict[str, Any]`):
+                Additional parameters from which to initialize the processor object.
+
+        Returns:
+            [`~processing_utils.ProcessingMixin`]: The processor object instantiated from those
+            parameters.
+        """
+        processor_dict = processor_dict.copy()
+        return_unused_kwargs = kwargs.pop("return_unused_kwargs", False)
+
+        # Unlike image processors or feature extractors whose `__init__` accept `kwargs`, processor don't have `kwargs`.
+        # We have to pop up some unused (but specific) arguments to make it work.
+        if "processor_class" in processor_dict:
+            del processor_dict["processor_class"]
+
+        if "auto_map" in processor_dict:
+            del processor_dict["auto_map"]
+
+        processor = cls(*args, **processor_dict)
+
+        # Update processor with kwargs if needed
+        to_remove = []
+        for key, value in kwargs.items():
+            if hasattr(processor, key):
+                setattr(processor, key, value)
+                to_remove.append(key)
+        for key in to_remove:
+            kwargs.pop(key, None)
```

Comment:
```suggestion
        # Update processor with kwargs if needed
        to_remove = []
        for keyin inspect.signature(self.__init__) :
            if key in kwargs:
                setattr(processor, key, kwargs.pop("key))
                to_remove.append(key)
```
should be simpler

Reply:
I would try to avoid assuming that any key in `inspect.signature(self.__init__)` would be an attribute in `processor`.
For `processor`, yes, that's what we should maintain, but I don't want to the code block here to `assume` this.

Second, if a key in `kwargs` that is not in `inspect.signature(self.__init__)`, it woon't be added to `to_remove` with the suggest change, which we should add it to `to_remove`.

Reply:
The code is overly compilcated to do something quite simple

Reply:
I disagree this is is overly compilcated  compared to what has been suggested, which also introduce unnecessary assumption of  `inspect.signature(self.__init__)`  and `processor`'s attributes.

And actually that code block is that small and direct, I don't understand why it is complicated - it's almost trivial to understand.

I can change it to manipulate the keys of `kwargs` and `processor.to_dict()` as two sets, but there is no real benefit IMO, as it again has to `processor.to_dict()` as well as `set()` construct as dict's `key()` are not set immediately.

Reply:
OK, let me try one thing

Reply:
Is the last commit ok

```python
        for key in set(kwargs.keys()):
            if hasattr(processor, key):
                setattr(processor, key, kwargs.pop(key))
```